components/Themes/src/MDCSemanticColorScheme.h

```diff
@@ -0,0 +1,80 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import "MDCColorScheme.h"
+
+/**
+ A simple color scheme that provides more semantic context for the colors it uses. Unlike
+ @c MDCColorScheme, there are no optional properties, so more consistent theming can be applied.
+ */
+@protocol MDCSemanticColorScheming <MDCColorScheme>
+
+/**
+ Displayed most frequently across your app.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *primaryColor;
+
+/**
+ A tonal variation of primary color.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *primaryColorLightVariant;
+
+/**
+ A tonal variation of primary color.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *primaryColorDarkVariant;
+
+/**
+ Accents select parts of your UI.
+ */
+@property(nullable, readonly, nonatomic) UIColor *secondaryColor;
+
+/**
+ The color used to indicate error status. Defaults to a red color.
+*/
+@property(nonnull, readonly, nonatomic) UIColor *errorColor;
+
+@end
+
+@interface MDCSemanticColorScheme : NSObject <MDCSemanticColorScheming>
+
+#pragma mark - MDCColorScheme compatibility
+
+// Bound to @c primaryColorLightVariant
+@property(nonatomic, strong, nonnull, readonly) UIColor *primaryLightColor;
+
+// Bound to @c primaryColorDarkVariant
+@property(nonatomic, strong, nonnull, readonly) UIColor *primaryDarkColor;
+
+#pragma mark - Initializers
+
+/**
+ Creates a new color scheme with the provided color parameters.
+
+ @param primaryColor Displayed most frequently across your app.
+ @param primaryColorLightVariant A tonal variation of @c primaryColor.
+ @param primaryColorDarkVariant A tonal variation of @c primaryColor.
+ @param secondaryColor Accents select parts of your UI. If @c nil, defaults to @c primaryColor.
+ @param errorColor The color used to indicate error status. If @c nil, defaults to a red color.
+*/
+- (nonnull instancetype)initWithPrimaryColor:(nonnull UIColor *)primaryColor
+                    primaryColorLightVariant:(nonnull UIColor *)primaryColorLightVariant
+                     primaryColorDarkVariant:(nonnull UIColor *)primaryColorDarkVariant
+                              secondaryColor:(nullable UIColor *)secondaryColor
+                                  errorColor:(nullable UIColor *)errorColor;
+
```

Comment:
Expose all of the properties on this class as writeable as well. It should be easy for someone to change the properties after initialization if they so desire (e.g. someone creates a Baseline color scheme initialized with the purples and then changes only the secondary color).

Reply:
Consider:
initWithPrimary:  // Calculate a sane secondary, error, etc.

I haven't seen this method of using nullable parameters as optionals before, typically I would expect all the parameters to be nonnull and there to be convenience constructors all wired into the designated init.

Reply:
We are intentionally not doing any code-side calculation of colors in the short term. That logic will live solely in the color tool.

Reply:
Regarding `readwrite` properties, do we have a specific use case where a client will need to modify a scheme once it has been created?  It seems to me that creating a new scheme as a variant of another is pretty easy.

```objectivec
  MDCSemanticColorScheme *colorScheme =
      [[MDCSemanticColorScheme alloc] initWithPrimaryColor:oldScheme.primaryColor
                                  primaryColorLightVariant:oldScheme.primaryColorLightVariant
                                   primaryColorDarkVariant:oldScheme.primaryColorDarkVariant
                                            secondaryColor:UIColor.magenta
                                                errorColor:[UIColor.red colorWithAlpha:0.5]];
```

Reply:
The primary use case I'm aware of is someone creating a Baseline color scheme and mutating its secondaryColor.

Reply:
Spoke offline with @romoore and agreed that the property setter case may not be particularly important and that it may be sufficient to just have a designated initializer in the first pass of the API.